jetbrains-core/src/software/aws/toolkits/jetbrains/utils/execution/steps/StepExecutor.kt

```diff
@@ -0,0 +1,127 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.utils.execution.steps
+
+import com.intellij.build.BuildDescriptor
+import com.intellij.build.BuildProgressListener
+import com.intellij.build.BuildViewManager
+import com.intellij.build.DefaultBuildDescriptor
+import com.intellij.build.events.impl.FailureResultImpl
+import com.intellij.build.events.impl.FinishBuildEventImpl
+import com.intellij.build.events.impl.StartBuildEventImpl
+import com.intellij.build.events.impl.SuccessResultImpl
+import com.intellij.execution.process.ProcessHandler
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.service
+import com.intellij.openapi.progress.ProcessCanceledException
+import com.intellij.openapi.project.Project
+import software.aws.toolkits.resources.message
+import java.io.OutputStream
+
+/**
+ * Executor for a tree of [Step]
+ */
+class StepExecutor(
+    private val project: Project,
+    private val title: String,
+    private val rootStep: Step,
+    private val uniqueId: String,
+    private val progressListener: BuildProgressListener = project.service<BuildViewManager>()
+) {
+    fun startExecution(): ProcessHandler {
+        val context = Context(project)
+        val processHandler = DummyProcessHandler(context)
+        val descriptor = DefaultBuildDescriptor(
+            uniqueId,
+            title,
+            "/unused/location",
```

Comment:
(since this calls into java) nit: `/unused/working/directory`

Reply:
changed